### PR TITLE
update actions version

### PR DESCRIPTION
**Pull request by zacharyburnett**
### Description

This pull request is to address ...

Fixes #<Issue Number>

### CODE REVIEW

**Comment by pllim** [comment]
What would be more future-proof is to instead submit a PR with `dependabot.yml` and let the bot do all the upgrades. Are you willing to do that? Thanks!

File: .github/workflows/ci_workflows.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - name: Check base branch
-      uses: actions/github-script@v6
+      uses: actions/github-script@v7
```
Comment (pllim) [inline .github/workflows/ci_workflows.yml:24]: I don't remember if github-script from v6 to v7 requires also other updates to go with it. You might have to check the change log.

**Comment by zacharyburnett** [comment]
> What would be more future-proof is to instead submit a PR with `dependabot.yml` and let the bot do all the upgrades. Are you willing to do that? Thanks!

Yes I think that's a great idea. I'll write up a PR

**Comment by pllim** [comment]
Thanks! I will close this without merge then so we can test to see if the bot works when it exists.

**Comment by pllim** [comment]
xref #377

**Pull request closed without merging**